File: src/main/java/io/playground/orderservice/application/order/port/OrderAmountPersistencePort.java
package io.playground.orderservice.application.order.port;

import io.playground.orderservice.application.order.dto.OrderAmountInfo;
import io.playground.orderservice.domain.order.OrderAmount;

import java.util.Optional;

public interface OrderAmountPersistencePort {
    Optional<OrderAmountInfo> findOrderAmountInfoByOrderExternalId(String orderExternalId);

    OrderAmount save(OrderAmount orderAmount);
}

File: src/main/kotlin/io/playground/orderservice/application/order/port/OrderAmountPersistencePort.kt
package io.playground.orderservice.application.order.port

import io.playground.orderservice.application.order.dto.OrderAmountInfo
import io.playground.orderservice.domain.order.OrderAmount
import java.util.Optional

interface OrderAmountPersistencePort {
    fun findOrderAmountInfoByOrderExternalId(orderExternalId: String): Optional<OrderAmountInfo>

    fun save(orderAmount: OrderAmount): OrderAmount
}
